public/app/plugins/datasource/grafana/components/AnnotationQueryEditor.tsx

```diff
@@ -66,49 +90,86 @@ export default function AnnotationQueryEditor({ query, onChange }: Props) {
       tags,
     });
 
+  const onQueryTypeChange = (newValue: SelectableValue<GrafanaQueryType>) => {
+    const newQuery: GrafanaAnnotationQuery = { ...annotationQuery, queryType: newValue.value! };
+    if (newQuery.queryType === GrafanaQueryType.TimeRegions) {
+      if (!newQuery.timeRegion) {
+        newQuery.timeRegion = {
+          timezone: defaultTimezone,
+        };
+      }
+    } else {
+      delete newQuery.timeRegion;
+    }
+
+    onChange(newQuery);
+  };
+  const onTimeRegionChange = (timeRegion?: TimeRegionConfig) => {
+    onChange({
+      ...annotationQuery,
+      timeRegion,
+    });
+  };
+
   return (
     <FieldSet className={styles.container}>
-      <Field label="Filter by">
+      <Field label="Query type">
         <Select
-          inputId="grafana-annotations__filter-by"
-          options={annotationTypes}
-          value={type}
-          onChange={onFilterByChange}
+          inputId="grafana-annotations__query-type"
```

Comment:
thoughts on creating selectors for newly added inputIds in this file? -> even if we don't write e2e test as part of this PR and instead make a future PR with those having these already as selectors would help with that

Reply:
I've created an [issue](https://github.com/grafana/grafana/issues/66887) to update e2es for annotations. If I see selectors I tend to assume they're used in a test already so I'd prefer to create them at the same time.